### PR TITLE
Add specification documents

### PR DESCRIPTION
**Pull request by hal-shu-sato**
## Description

- Move v1 files to the `v1` directory.
- Create a symbolic link for v1.
- Add specification documents for 3 versions.

## Checklist:

- [ ] ~~Updated the modification date in `mod.xml`.~~

### CODE REVIEW

File: v3/SPECIFICATION.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+- **\[program\].releases**: プログラムのリリース一覧（ReleaseData[]）
+  - **ReleaseData.version**: そのリリースのバージョン（String）
+  - **ReleaseData.url**: リリースされたアーカイブのURL（String）
+  - **ReleaseData.integrity**: ハッシュの一覧（Array）
+    - **ReleaseData.integrity.archive**: アーカイブファイルのハッシュ（String）
+    - **ReleaseData.integrity.file**: パッケージファイルのハッシュ一覧（FileIntegrityData[]）
+      - **FileIntegrityData.target**: ハッシュの該当ファイル（String）
+      - **FileIntegrityData.hash**: ハッシュ（`sha384-`などから始まるString）
+
+## [packages_list.xml](./data/packages_list.xml)
```
Comment (mitosagi) [inline v3/SPECIFICATION.md:43]:
```suggestion
## [packages.json](./packages.json)
```
Similarly, please change the other links in v3/SPECIFICATION.md.

**Review by mitosagi — CHANGES_REQUESTED** [review]
I would like to confirm one thing.

**Comment by hal-shu-sato** [comment]
I fixed it.

In my search, to enable symlinks in Git, we have to set `core.symlinks` to `true`.

**Comment by hal-shu-sato** [comment]
Symbolic links were added to maintain the availability of v1.
However, since we are no longer supporting it, we may not have to take that step.
Also, jsDelivr does not seem to be supported.

**Comment by mitosagi** [comment]
What about deleting v1 files?

**Comment by hal-shu-sato** [comment]
I have a desire to keep the data files as legacy.
I am now thinking that I may remove Symlink.

**Comment by hal-shu-sato** [comment]
By the way, I notice that I made a mistake regarding the necessary element in `SPECIFICATION.md`.
I will fix it.

**Review by mitosagi — APPROVED** [review]
Hopefully the documentation will help new participants.

Symbolic links can be removed at your own discretion.

**Comment by hal-shu-sato** [comment]
I removed the symlink and rebased.

**Pull request merged**